subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/initialization/ConfigurationCacheStartParameter.kt

```diff
@@ -72,7 +74,7 @@ class ConfigurationCacheStartParameter(
 
     @Suppress("DEPRECATION")
     val settingsFile: File?
-        get() = startParameter.settingsFile
+        get() = DeprecationLogger.whileDisabled(Factory { startParameter.settingsFile })
```

Comment:
Maybe it's a stupid question, but if we add an advice to the deprecation warning about what to do instead, then why don't we take our own medicine instead of hacking the deprecation warning out of existence for our own usages? Does that mean that our advice is not that easy to follow?